addons/purchase/models/purchase.py

```diff
@@ -917,10 +917,23 @@ def _run(self):
 
     @api.multi
     def _check(self):
-        if self.purchase_line_id:
+        if self.rule_id.action == 'buy':
+            # In case Phantom BoM splits only into procurements
             if not self.move_ids:
+                if self.purchase_line_id and self.purchase_line_id.order_id.state not in ('purchase', 'done', 'cancel'):
+                    return False
+                else:
+                    return True
+            move_all_done_or_cancel = all(move.state in ['done', 'cancel'] for move in self.move_ids)
+            move_all_cancel = all(move.state == 'cancel' for move in self.move_ids)
```

Comment:
We may reduce search time by:
move_all_cancel = all(move.state == 'cancel' for move in move_all_done_or_cancel)